cin_validator/rules/cin2022_23/rule_8897.py

```diff
@@ -0,0 +1,176 @@
+from typing import Mapping
+
+import pandas as pd
+
+from cin_validator.rule_engine import CINTable, RuleContext, rule_definition
+from cin_validator.test_engine import run_rule
+from cin_validator.utils import make_census_period
+
+# Get tables and columns of interest from the CINTable object defined in rule_engine/__api.py
+
+Assessments = CINTable.Assessments
+AssessmentAuthorisationDate = Assessments.AssessmentAuthorisationDate
+AssessmentFactors = Assessments.AssessmentFactors
+LAchildID = Assessments.LAchildID
+
+Header = CINTable.Header
+ReferenceDate = Header.ReferenceDate
+
+# define characteristics of rule
+@rule_definition(
+    # write the rule code here, in place of 8500
+    code=8897,
+    # replace ChildIdentifiers with the value in the module column of the excel sheet corresponding to this rule .
+    module=CINTable.Assessments,
+    # replace the message with the corresponding value for this rule, gotten from the excel sheet.
+    message="Parental or child factors at assessment information is missing from a completed assessment",
+    # The column names tend to be the words within the < > signs in the github issue description.
+    affected_fields=[AssessmentAuthorisationDate, AssessmentFactors],
+)
+def validate(
+    data_container: Mapping[CINTable, pd.DataFrame], rule_context: RuleContext
+):
+    # PREPARING DATA
+
+    # Replace ChildIdentifiers with the name of the table you need.
+    df = data_container[Assessments]
+    # Before you begin, rename the index so that the initial row positions can be kept intact.
+    df.index.name = "ROW_ID"
+
+    # lOGIC
+    # Implement rule logic as described by the Github issue.
+    # Put the description as a comment above the implementation as shown.
+
+    # Where present, if <AssessmentAuthorisationDate> (N00160) is on or after [Start_Of_Census_Year] then one or more <AssessmentFactors> (N00181) must be present within the same assessment module and must be a valid code
+    # Get collection period
+    header = data_container[Header]
+    ref_date_series = header[ReferenceDate]
+    collection_start, collection_end = make_census_period(ref_date_series)
+    condition1 = (df[AssessmentAuthorisationDate] >= collection_start) & (
+        df[AssessmentAuthorisationDate].notna()
+    )
+    condition2 = df[AssessmentFactors].isna()
```

Comment:
Hi Wendy, this rule needs you to check both if AssessmentFactor is present AND if it's a valid code. The codes can be found here: https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/1025195/Children_in_need_census_2022_to_2023_guide.pdf

To test if particular column on a particular row contains a valid code, I'd first make a list of valid codes, then use the .isin(codes_list_name) method on the column of the df, just as you have used .isna() and .notna() for conditions 1 and 2. If you feel you don't have time to do this, I'm happy to do it and send you an email with pictures explaining what I've done.